integration-tests/src/tests/client/features/multinode_stateless_validators.rs

```diff
@@ -0,0 +1,674 @@
+    ClientDelayedActions(TestLoopDelayedActionEvent<ClientActorInner>),
+    /// Allows delayed actions to be posted, as if ShardsManagerActor scheduled them, e.g. timers.
+    ShardsManagerDelayedActions(TestLoopDelayedActionEvent<ShardsManagerActor>),
+    /// Allows delayed actions to be posted, as if SyncJobsActor scheduled them, e.g. timers.
+    SyncJobsDelayedActions(TestLoopDelayedActionEvent<SyncJobsActor>),
+
+    /// Message that the network layer sends to the client.
+    ClientEventFromNetwork(ClientSenderForNetworkMessage),
+    /// Message that the client sends to the client itself.
+    ClientEventFromClient(ClientSenderForClientMessage),
+    /// Message that the SyncJobs component sends to the client.
+    ClientEventFromSyncJobs(ClientSenderForSyncJobsMessage),
+    /// Message that the ShardsManager component sends to the client.
+    ClientEventFromShardsManager(ShardsManagerResponse),
+    /// Message that the state sync adapter sends to the client.
+    ClientEventFromStateSyncAdapter(SyncMessage),
+
+    /// Message that the client sends to the SyncJobs component.
+    SyncJobsEventFromClient(SyncJobsSenderForClientMessage),
+
+    /// Message that the client sends to the SyncActor component.
+    SyncActorEventFromClient((ShardUId, SyncMessage)),
+    /// Message that the network sends to the SyncActor component.
+    SyncActorEventFromNetwork((ShardUId, StateSyncResponse)),
+
+    /// Message that the client sends to the ShardsManager component.
+    ShardsManagerRequestFromClient(ShardsManagerRequestFromClient),
+    /// Message that the network layer sends to the ShardsManager component.
+    ShardsManagerRequestFromNetwork(ShardsManagerRequestFromNetwork),
+
+    /// Message that the client sends to StateSnapshotActor.
+    StateSnapshotRequestFromClient(StateSnapshotSenderForClientMessage),
+    /// Message that the StateSnapshotActor sends to itself.
+    StateSnapshotRequestFromStateSnapshot(StateSnapshotSenderForStateSnapshotMessage),
+
+    /// Outgoing network message that is sent by any of the components of this node.
+    OutgoingNetworkMessage(PeerManagerMessageRequest),
+    /// Same as OutgoingNetworkMessage, but of the variant that requests a response.
+    OutgoingNetworkMessageForResult(
+        MessageWithCallback<PeerManagerMessageRequest, PeerManagerMessageResponse>,
+    ),
+    /// Calls to the network component to set chain info.
+    SetChainInfo(SetChainInfo),
+    /// Message from Client to PartialWitnessActor.
+    PartialWitnessSenderForClient(PartialWitnessSenderForClientMessage),
+    /// Message from Network to PartialWitnessActor.
+    PartialWitnessSenderForNetwork(PartialWitnessSenderForNetworkMessage),
+    /// Message from PartialWitnessActor to Client.
+    ClientSenderForPartialWitness(ClientSenderForPartialWitnessMessage),
+}
+
+const ONE_NEAR: u128 = 1_000_000_000_000_000_000_000_000;
+
+const NUM_ACCOUNTS: usize = 20;
+const NUM_VALIDATORS: usize = 8;
```

Comment:
nit: move below, set `const NUM_VALIDATORS: usize = NUM_BLOCK_AND_CHUNK_PRODUCERS + NUM_CHUNK_VALIDATORS_ONLY;` instead of assertion?